src/mono/wasm/debugger/BrowserDebugProxy/JObjectValueCreator.cs

```diff
@@ -265,22 +265,51 @@ async Task<string> GetNullObjectClassName()
             return className;
         }
     }
-
+    private async Task<string> ToString(List<int> typeIds, int objectId, CancellationToken token)
+    {
+        string description = "";
+        try {
+            foreach (var typeId in typeIds)
+            {
+                var typeInfo = await _sdbAgent.GetTypeInfo(typeId, token);
+                if (typeInfo != null && typeInfo.Name != "object")
+                {
+                    MethodInfo methodInfo = typeInfo.Info.Methods.FirstOrDefault(m => m.Name == "ToString");
+                    if (methodInfo != null)
+                    {
+                        int[] methodIds = await _sdbAgent.GetMethodIdsByName(typeId, "ToString", token);
+                        if (methodIds != null && methodIds.Length > 0)
+                        {
+                            var toString = await _sdbAgent.InvokeMethod(objectId, methodIds[0], isValueType: false, token);
+                            if (toString["value"]?["value"] != null)
+                                description = toString["value"]?["value"].Value<string>();
+                        }
+                        break;
+                    }
+                }
+            }
+        }
+        catch (Exception e)
+        {
+            _logger.LogDebug($"Error while evaluating ToString method: {e}");
+        }
+        return description;
+    }
     private async Task<JObject> ReadAsObjectValue(MonoBinaryReader retDebuggerCmdReader, int typeIdFromAttribute, bool forDebuggerDisplayAttribute, CancellationToken token)
     {
         var objectId = retDebuggerCmdReader.ReadInt32();
-        var type_id = await _sdbAgent.GetTypeIdsForObject(objectId, false, token);
-        string className = await _sdbAgent.GetTypeName(type_id[0], token);
+        var typeIds = await _sdbAgent.GetTypeIdsForObject(objectId, true, token);
```

Comment:
use parameter name for the bool. And 👍 on the local rename!

Reply:
done.